### PR TITLE
Replace `strtobool()` #10460

### LINKED ISSUE #10460: Python 3.12 compat: Replace `strtobool()`
The boolean datatype uses `strtobool()` in a couple places, which comes from earlier versions of `setuptools`, which we were pulling in as a transitive dependency from `mapbox-vector-tile`.

Noticed on a fresh install of Python 3.12 (which no longer has `distutils` anyway).

We can write a teeny replacement to keep the existing functionality. Suggest upgrading `mapbox-vector-tile` so that fresh installs don't pull in `setuptools` and the failure can be seen more clearly.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
Depends on archesproject/arches#10456

On python 3.12, Arches had an undeclared dependency on a util (`strtobool()`) from older versions of `setuptools` which still vendored distutils, since `distutils` is no longer in the stdlib. Instead just reimplement the functionality to avoid pulling in setuptools or anything else as a dep.

~Bumps `mapbox-vector-tile` to a version that doesn't pull in `setuptools` as a dependency (and a version that explicitly supports python 3.11).~ EDIT: removes it altogether.

### Issues Solved
Closes #10460

### Checklist

-   [x] Unit tests pass locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/utils/datatype_tests.py [hunk 1]
```diff
@@ -26,6 +26,36 @@
...
+class BooleanDataTypeTests(ArchesTestCase):
+    def test_validate(self):
+        boolean = DataTypeFactory().get_instance("boolean")
+
+        for good in ["true", "false", "yes", "no", None]:
+            with self.subTest(input=good):
+                no_errors = boolean.validate(good)
+                self.assertEqual(len(no_errors), 0)
+
+        for bad in ["garbage", "True", "False", "None"]:
```
Comment (jacobtylerwalls) [inline tests/utils/datatype_tests.py:38]: We reject upper cased True/False, so I made sure to surface that here.

File: arches/app/utils/string_utils.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def str_to_bool(value):
+    match value:
+        case "y" | "yes" | "t" | "true" | "on" | "1":
+            return True
+        case "n" | "no" | "f" | "false" | "off" | "0":
+            return False
+    raise ValueError
```
Comment (jacobtylerwalls) [inline arches/app/utils/string_utils.py:7]: Reference: https://docs.python.org/3.11/distutils/apiref.html?highlight=strtobool#distutils.util.strtobool

File: arches/install/requirements.txt [hunk 3]
```diff
@@ -7,7 +7,7 @@ django-guardian==2.4.0
 python-memcached==1.59
 celery==5.3.6
 django-celery-results==2.5.1
-mapbox-vector-tile==1.2.0
+mapbox-vector-tile==2.0.1
```
Comment (chiatt) [inline arches/install/requirements.txt:10]: I wonder if we need this package at all anymore.

Comment (jacobtylerwalls) [reply arches/install/requirements.txt:10]: Oh good call. Looks like TileStache was removed in arches 5. Maybe it was here for that?

Comment (chiatt) [reply arches/install/requirements.txt]: Yeah - that must be the case. Arches used to ship with a tile server using TileStache, but now we just let people implement their own and proxy tile requests through Arches.

**Review by chiatt — APPROVED** [review]

**Pull request merged**